kvdb/postgres/fixture.go

```diff
@@ -8,86 +8,166 @@ import (
 	"database/sql"
 	"encoding/hex"
 	"fmt"
+	"strconv"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/btcsuite/btcwallet/walletdb"
-	embeddedpostgres "github.com/fergusstrange/embedded-postgres"
+	_ "github.com/lib/pq"
 	"github.com/lightningnetwork/lnd/kvdb/sqlbase"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/stretchr/testify/require"
 )
 
 const (
-	testDsnTemplate = "postgres://postgres:postgres@localhost:9876/%v?sslmode=disable"
+	testPgUser  = "test"
+	testPgPass  = "test"
+	testPgDb    = "test"
+	PostgresTag = "11"
+
+	testDsnTemplate = "postgres://test:test@%s:%d/%s?sslmode=disable"
 	prefix          = "test"
+
+	testMaxConnections = 200
 )
 
-func getTestDsn(dbName string) string {
-	return fmt.Sprintf(testDsnTemplate, dbName)
-}
+var (
+	// DefaultExpiry is the default expiry time for the test fixture. This
+	// means a single unit test is not allowed to run more than 3 hours
+	// (which is the same duration as the default test timeout).
+	DefaultExpiry = 180 * time.Minute
+)
 
-var testPostgres *embeddedpostgres.EmbeddedPostgres
+func getTestDsn(host string, port int, dbName string) string {
+	return fmt.Sprintf(testDsnTemplate, host, port, dbName)
+}
 
-const testMaxConnections = 200
+var testPostgres *testPgFixture
```

Comment:
This only is being used for the `make unit tags=kvdb_postgres` tests. In the normal `make unit` and `make unit-race` this shouldn't be used. So it could explain the failures in the Postgres test, but not in the others...